sqlx-mysql/src/connection/infile.rs

```diff
@@ -0,0 +1,211 @@
+//! Support for `LOAD DATA LOCAL INFILE` statements
+//!
+//! This MySQL feature allows the client to send a local file to the server, which is then
+//! loaded into a table. This should be faster than sending the data row-by-row.
+//!
+//! # Example
+//! ```rust,no_run
+//! use sqlx::mysql::infile::{MySqlExecutorInfileExt, LocalInfileHandler};
+//!
+//! #[tokio::main]
+//! async fn main() -> Result<(), sqlx::Error> {
+//!     let pool = sqlx::mysql::MySqlPool::connect("mysql://root:password@localhost:3306/sqlx")
+//!         .await?;
+//!
+//!     let res = pool
+//!         .local_infile_statement(
+//!             "LOAD DATA LOCAL INFILE 'dummy' INTO TABLE testje",
+//!             LocalInfileHandler::new(|filename, stream| {
+//!                 assert_eq!(filename, b"dummy");
+//!                 Box::pin(async move {
+//!                     stream.write(b"1\n2\n3\n4\n5\n6\n7\n8\n9\n10").await?;
+//!                     Ok(())
+//!                 })
+//!             }),
+//!         )
```

Comment:
I'm not really a fan of this API design. I'd prefer it to look more like the API for `COPY FROM STDIN` in `sqlx-postgres`: 
* https://docs.rs/sqlx-postgres/latest/sqlx_postgres/struct.PgConnection.html#method.copy_in_raw
* https://docs.rs/sqlx-postgres/latest/sqlx_postgres/struct.PgCopyIn.html

This would probably mean not using `MySqlConnection::run()` and using a more specialized routine instead, but it would simplify the usage a lot.

As for finding out what the filename is, I'd just add a getter to `InfileDataStream`. 90% of the time the filename is either going to be a dummy value or already known to the application.

Reply:
Interesting, I didn't think to look at existing postgres functionality. Thanks for giving me the pointers, I'll see if I can rework it to match

Reply:
I changed the API, and the new test succeeds. I like the new API more too